templates/support/OAuthAccessTokenManager.swift.hbs

```diff
@@ -0,0 +1,126 @@
+//  
+//  {{>frag/generatedBy}}
+//
+
+/** A token manager that will group refresh attempts to prevent concurrent token requests */
+import Foundation
+
+actor OAuthAccessTokenManager {
+    
+    var accessToken: OAuthToken? {
+        didSet {
+            restartRefreshTimer()
+        }
+    }
+    
+    private let refreshTokenURL: URL?
+    private let automaticallyRefreshTokensThisManySecondsEarly: TimeInterval?
```

Comment:
Do we need to have "seconds" in this name everywhere when it's a `TimeInterval`? I think a `TimeInterval` is explicit about its units? And we can probably remove the word "automatically" and just call it "preemptiveRefresh" or "preemptiveAccessTokenRefresh"? Or something like that?